hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/safemode/TestSCMSafeModeManager.java

```diff
@@ -215,34 +216,22 @@ private OzoneConfiguration createConf(double healthyPercent,
     return conf;
   }
 
-  @Test
-  public void testSafeModeExitRuleWithPipelineAvailabilityCheck1()
-      throws Exception {
-    testSafeModeExitRuleWithPipelineAvailabilityCheck(100, 30, 8, 0.90, 1);
-  }
-
-  @Test
-  public void testSafeModeExitRuleWithPipelineAvailabilityCheck2()
-      throws Exception {
-    testSafeModeExitRuleWithPipelineAvailabilityCheck(100, 90, 22, 0.10, 0.9);
-  }
-
-  @Test
-  public void testSafeModeExitRuleWithPipelineAvailabilityCheck3()
-      throws Exception {
-    testSafeModeExitRuleWithPipelineAvailabilityCheck(100, 30, 8, 0, 0.9);
-  }
-
-  @Test
-  public void testSafeModeExitRuleWithPipelineAvailabilityCheck4()
-      throws Exception {
-    testSafeModeExitRuleWithPipelineAvailabilityCheck(100, 90, 22, 0, 0);
+  private static Stream<Arguments> testCaseForSafeModeExitRuleWithPipelineAvailabilityCheck() {
+    return Stream.of(
+        Arguments.of(100, 30, 8, 0.90, 1),
+        Arguments.of(100, 90, 22, 0.10, 0.9),
+        Arguments.of(100, 30, 8, 0, 0.9),
+        Arguments.of(100, 90, 22, 0, 0),
+        Arguments.of(100, 90, 22, 0, 0.5)
+    );
   }
 
-  @Test
-  public void testSafeModeExitRuleWithPipelineAvailabilityCheck5()
-      throws Exception {
-    testSafeModeExitRuleWithPipelineAvailabilityCheck(100, 90, 22, 0, 0.5);
+  @ParameterizedTest
+  @MethodSource("testCaseForSafeModeExitRuleWithPipelineAvailabilityCheck")
+  public void testSafeModeExitRuleWithPipelineAvailabilityCheck1(int containerCount, int nodeCount, int pipelineCount,
+      double healthyPipelinePercent, double oneReplicaPercent) throws Exception {
+    testSafeModeExitRuleWithPipelineAvailabilityCheck(containerCount,
+        nodeCount, pipelineCount, healthyPipelinePercent, oneReplicaPercent);
```

Comment:
Let's use the parameterization directly in `testSafeModeExitRuleWithPipelineAvailabilityCheck` instead of `testSafeModeExitRuleWithPipelineAvailabilityCheck1` which has the same function signatures.

Reply:
Understood! Thank you for the review. The `testSafeModeExitRuleWithPipelineAvailabilityCheck1` has been removed.